src/sentry_envelope.c

```diff
@@ -234,6 +255,45 @@ sentry__envelope_add_event(sentry_envelope_t *envelope, sentry_value_t event)
     return item;
 }
 
+sentry_envelope_item_t *
+sentry__envelope_add_transaction(
```

Comment:
a majority of this is just copypasted from `sentry__envelope_add_event`, with a few differences:
- this sets a different item header value
- this sets a `sent_at` header value

this is duplicated to minimize impact on the event implementation, and to mostly follow the pattern established by other SDKs (they also have separate functions to add different types of events to an envelope). 

of course, this misses out on some of the nice perks that the other SDKs gain from having this pattern, but i think minimizing the impact of this change is enough to justify this for now. open to changing my opinion on this, though. if merged into `sentry__envelope_add_event`, this would only introduce 1 or 2 additional if statements (for now).

Reply:
I think the `sent_at` value is applicable to all envelopes? Should we rather add this really at "send" time, as opposed to "adding item to envelope" time?

Also, maybe split this into a `sentry__envelope_add_value(sentry_value_t value, const char *type)` that rather takes the `type` header value explicitly?

Reply:
> I think the sent_at value is applicable to all envelopes? Should we rather add this really at "send" time, as opposed to "adding item to envelope" time?

i wouldn't be opposed to this, although it looks like all of the code paths that use this (and `sentry__envelope_add_event`) send the envelope immediately afterwards. this means that any inaccuracies or timing offsets in`sent_at` would be caused by things like adding/ending the current session, or the amount of time it takes to push it completely through the transport.

what do you think of the idea of documenting this somewhere, and if somebody notices any issues with the `sent_at` timestamps we can set the header a little bit closer to when it's actually sent? something to consider is that events being sent by the native SDK right now (_not_ transactions) do not have the `sent_at` header. i'm not sure if it's because the header precedes this implementation, or if there's some sort of subtle timing issue that lead to this omission.

> maybe split this into a sentry__envelope_add_value(sentry_value_t value, const char *type) that rather takes the type header value explicitly?

this is a good idea and i'll give this a try in a follow-up PR, if that's alright with you. it should be a simple change, hopefully.

Reply:
Sure, go for it in a followup. And the suggestion for `sent_at` seems to be moot as well. I don’t think we need that much precision after all :-D